excel_graph_parser/parser.py

```diff
@@ -0,0 +1,214 @@
+                        input_cat_range = series.cat.numRef.f
+                        input_cat_format = series.cat.numRef.numCache.formatCode
+
+                input_val_range = series.val.numRef.f
+                input_val_format = series.val.numRef.numCache.formatCode
+
+            input_cat_format = None if input_cat_format == "General" else input_cat_format
+            input_val_format = None if input_val_format == "General" else input_val_format
+
+            # category_axis_data
+            chart_sheet_name = (
+                input_cat_range
+                .replace("(", "")
+                .replace(")", "")
+                .replace("'", "")
+                .split(sep="!")[0]
+            )
+            chart_cat_range = (
+                input_cat_range
+                .replace('(', '')
+                .replace(')', '')
+                .replace("'", "")
+                .replace(f"{chart_sheet_name}!", "")
+                .replace('$', "")
+            )
+
+            cat_range_start = chart_cat_range.split(",")[0]
+            cat_range_end = chart_cat_range.split(",")[-1] if "," in chart_cat_range else chart_cat_range
+            cat_data = []
+            for element in wb[chart_sheet_name][f"{cat_range_start}:{cat_range_end}"]:
+                cat_data = [e.value for e in element if type(e) == Cell]
+
+            # value_axis_data
+            chart_sheet_name = (
+                input_val_range
+                .replace('(', '')
+                .replace(')', '')
+                .replace("'", "")
+                .split(sep="!")[0]
+            )
+            chart_val_range = (
+                input_val_range
+                .replace('(', '')
+                .replace(')', '')
+                .replace("'", "")
+                .replace(f"{chart_sheet_name}!", "")
+            )
+            val_range_start = chart_val_range.split(",")[0]
+            val_range_end = chart_val_range.split(",")[-1] if "," in chart_val_range else chart_val_range
+            val_data = []
+            for element in wb[chart_sheet_name][f"{val_range_start}:{val_range_end}"]:
+                val_data = [e.value for e in element if type(e) == Cell]
+
+            series_name = series.tx.v if series.tx else None
+            ser = {
+                "category_axis_data": cat_data,
+                "value_axis_data": val_data,
+                "category_value_format": input_cat_format,
+                "values_value_format": input_val_format,
+                "series_name": series_name if series_name else None
+            }
+            series.append(ser)
+
+        chart_data = {
+            "chart_title": chart_title,
+            "chart_type": chart_type,
+            "x_axis_title": x_axis_title,
+            "y_axis_title": y_axis_title,
+            "series": series,
+        }
+
+        return chart_data
+
+    @staticmethod
+    def _create_plotly_figure(chart_data: dict) -> go.Figure:

```

Comment:
I think we should lift this outside of the parser. The parser is in charge of converting a spreadsheet into generic chart data.

Then the `create_plotly_figure_from_chart_data()` is a nice util on top of it.

Reply:
We could also rewrite this to `convert_chart_data_to_plotly_data() -> dict:` waarbij de `to.json()` all gecalled word.